static/js/copy_and_paste.js

```diff
@@ -96,22 +147,8 @@ function copy_handler() {
             return;
         }
 
-            // Construct a div for what we want to copy (div)
-        for (row = current_msg_list.get_row(start_id);
-             rows.id(row) <= end_id;
-             row = rows.next_visible(row)) {
-             if (row.prev().hasClass("message_header")) {
-                content = $('<div>').text(row.prev().text()
-                                            .replace(/\s+/g, " ")
-                                            .replace(/^\s/, "").replace(/\s$/, ""));
-                div.append($('<p>').append($('<strong>').text(content.text())));
-            }
-            message = current_msg_list.get(rows.id(row));
-            var message_firstp = $(message.content).slice(0, 1);
-            message_firstp.prepend(message.sender_full_name + ": ");
-            div.append(message_firstp);
-            div.append($(message.content).slice(1));
-        }
```

Comment:
For this first commit, it would have been better to do the refactor in a few commits (first ones that move code around without changing functionality, and then a commit to actually do the changes).  That makes things much more readable for code review.